ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -35,14 +44,125 @@ def _set_base_span_tags(
+                    elif block.get("type") == "image":
+                        # Store a placeholder for potentially enormous binary image data.
+                        input_messages.append({"content": "([IMAGE DETECTED])", "role": role})
+                    else:
+                        input_messages.append({"content": str(block), "role": role})
+
+        return input_messages
+
+    def _extract_output_message(self, response):
+        """Extract output messages from the stored response."""
+        output_messages = []
+        content = _get_attr(response, "content", None)
+        role = _get_attr(response, "role", "")
+
+        if isinstance(content, str):
+            return [{"content": self.trunc(content), "role": role}]
+
+        elif isinstance(content, list):
+            for completion in content:
+                text = _get_attr(completion, "text", None)
+                if isinstance(text, str):
+                    output_messages.append({"content": self.trunc(text), "role": role})
+        return output_messages
+
     def record_usage(self, span: Span, usage: Dict[str, Any]) -> None:
         if not usage:
             return
-        input_tokens = _get_attr(usage, "input_tokens", None)
-        output_tokens = _get_attr(usage, "output_tokens", None)
+        input_tokens = _get_attr(usage, "input_tokens", 0)
+        output_tokens = _get_attr(usage, "output_tokens", 0)
 
-        span.set_metric("anthropic.response.usage.input_tokens", input_tokens)
-        span.set_metric("anthropic.response.usage.output_tokens", output_tokens)
-
-        if input_tokens is not None and output_tokens is not None:
+        if input_tokens != 0:
+            span.set_metric("anthropic.response.usage.input_tokens", input_tokens)
+        if output_tokens != 0:
+            span.set_metric("anthropic.response.usage.output_tokens", output_tokens)
+        if input_tokens != 0 and output_tokens != 0:
             span.set_metric("anthropic.response.usage.total_tokens", input_tokens + output_tokens)
+
+    @classmethod
+    def _get_llmobs_metrics_tags(cls, span):
+        usage = {}
+        prompt_tokens = span.get_metric("anthropic.response.usage.input_tokens")
+        completion_tokens = span.get_metric("anthropic.response.usage.output_tokens")
+        total_tokens = span.get_metric("anthropic.response.usage.total_tokens")
+
+        if prompt_tokens is not None:
+            usage["prompt_tokens"] = prompt_tokens
+        if completion_tokens is not None:
+            usage["completion_tokens"] = completion_tokens
+        if total_tokens is not None:
+            usage["total_tokens"] = total_tokens
+        return usage
+
+
+def _get_attr(o: Any, attr: str, default: Any):
+    # Since our response may be a dict or object, convenience method
+    if isinstance(o, dict):
+        return o.get(attr, default)
+    else:
+        return getattr(o, attr, default)
```

Comment:
Is there a reason we have to use this method? From what I understand anthropic objects are always objects, in what cases are the objects dictionaries?

Reply:
This is required once we add streaming since we are building up the response into a dict

Reply:
Cool, let's keep this here then, and remove the duplicate function in `ddtrace.contrib.anthropic.utils.py`

Reply:
done